internal/provider/resource_jwks_url.go

```diff
@@ -0,0 +1,173 @@
+}
+
+func updateStateJwksUrl(d *schema.ResourceData, v neon.JWKS, role_names []string) error {
+	if err := d.Set("project_id", v.ProjectID); err != nil {
+		return err
+	}
+	if err := d.Set("jwks_url", v.JwksURL); err != nil {
+		return err
+	}
+	if err := d.Set("provider_name", v.ProviderName); err != nil {
+		return err
+	}
+	// todo - we don't get this back from the API, is using the provided values ok?
+	if err := d.Set("role_names", role_names); err != nil {
+		return err
+	}
+	if err := d.Set("branch_id", v.BranchID); err != nil {
+		return err
+	}
+	if err := d.Set("jwt_audience", v.JwtAudience); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceJwksUrlCreateRetry(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	return projectReadiness.Retry(resourceJwksUrlCreate, ctx, d, meta)
+}
+
+func resourceJwksUrlCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) error {
+	tflog.Trace(ctx, "created JWKS URL")
+
+	cfg := neon.AddProjectJWKSRequest{
+		JwksURL:      d.Get("jwks_url").(string),
+		ProviderName: d.Get("provider_name").(string),
+		RoleNames:    d.Get("role_names").([]string),
+		BranchID:     pointer(d.Get("branch_id").(string)),
+		JwtAudience:  pointer(d.Get("jwt_audience").(string)),
+	}
+
+	client := meta.(*neon.Client)
+	resp, err := client.AddProjectJWKS(d.Get("project_id").(string), cfg)
+	if err != nil {
+		return err
+	}
+	waitUnfinishedOperations(ctx, client, resp.OperationsResponse.Operations)
+
+	d.SetId(resp.Jwks.ID)
+
+	return updateStateJwksUrl(d, resp.JWKSResponse.Jwks, cfg.RoleNames)
+}
+
+func resourceJwksUrlReadRetry(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	return projectReadiness.Retry(resourceJwksUrlRead, ctx, d, meta)
+}
+
+func resourceJwksUrlRead(ctx context.Context, d *schema.ResourceData, meta interface{}) error {
+	tflog.Trace(ctx, "read JWKS URL")
+
+	projectID, _ := d.Get("project_id").(string)
+	id, _ := d.Get("id").(string)
+
+	resp, err := meta.(*neon.Client).GetProjectJWKS(projectID)
+	if err != nil {
+		return err
+	}
+
+	idx := slices.IndexFunc(resp.Jwks, func(j neon.JWKS) bool { return j.ID == id })
+	if idx < 0 {
+		return errors.New("JWKS URL resource was not found")
+	}
+	jwks := resp.Jwks[idx]
+
+	return updateStateJwksUrl(d, jwks, []string{})
+}
+
+func resourceJwksUrlDeleteRetry(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	return projectReadiness.Retry(resourceJwksUrlDelete, ctx, d, meta)
+}
+
+func resourceJwksUrlDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) error {
+	tflog.Trace(ctx, "delete JWKS URL")
+	client := meta.(*neon.Client)
+	resp, err := client.DeleteProjectJWKS(
+		d.Get("project_id").(string),
+		d.Get("id").(string),
+	)
+	if err != nil {
+		return err
+	}
+	//todo - the response object doesn't have an OperationsResponse attribute
+	waitUnfinishedOperations(ctx, client, resp.OperationsResponse.Operations)
```

Comment:
@kislerdm Noticed that the golang client doesn't return an `OperationsResponse` to wait for on this endpoint - wasn't sure if that was expected, or if there are simply no operations that we need to wait on.

Reply:
Let's consider operations running in sync if the response does not contains the slice of operations. No need to wait in such case.